core/chains/configs.go

```diff
@@ -0,0 +1,233 @@
+package chains
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/assets"
+)
+
+type (
+
+	// ChainSpecificConfig lists the config defaults specific to a particular chain ID
+	ChainSpecificConfig struct {
+		set                              bool
+		EnableLegacyJobPipeline          bool
+		EthBalanceMonitorBlockDelay      uint16
+		EthFinalityDepth                 uint
+		EthGasBumpThreshold              uint64
+		EthGasBumpWei                    big.Int
+		EthGasLimitDefault               uint64
+		EthGasLimitTransfer              uint64
+		EthGasPriceDefault               big.Int
+		EthHeadTrackerHistoryDepth       uint
+		EthHeadTrackerSamplingInterval   time.Duration
+		EthMaxGasPriceWei                big.Int
+		EthMaxInFlightTransactions       uint32
+		EthMaxQueuedTransactions         uint64
+		EthMinGasPriceWei                big.Int
+		EthTxResendAfterThreshold        time.Duration
+		GasUpdaterBatchSize              uint32
+		GasUpdaterBlockDelay             uint16
+		GasUpdaterBlockHistorySize       uint16
+		GasUpdaterEnabled                bool
+		LinkContractAddress              string
+		MinIncomingConfirmations         uint32
+		MinRequiredOutgoingConfirmations uint64
+		MinimumContractPayment           *assets.Link
+	}
+)
+
+// FallbackConfig represents the "base layer" of config defaults
+// It can be overridden on a per-chain basis and may be used if the chain is unknown
+var FallbackConfig ChainSpecificConfig
+
+func setConfigs() {
+	// --------------------------IMPORTANT---------------------------
+	// All config sets should "inherit" from FallbackConfig and overwrite
+	// fields as necessary. Do not create a new ChainSpecificConfig from
+	// scratch for a particular chain, since it may accidentally contain zero
+	// values.
+	// Be sure to copy and --not modify-- FallbackConfig!
+	// TODO: Warn if any of these are overridden by user-specified config vars
+	// See: https://app.clubhouse.io/chainlinklabs/story/11090/warn-if-nop-has-overridden-any-default-config-var
+	// TODO: We should probably move these to TOML or JSON files
+	// See: https://app.clubhouse.io/chainlinklabs/story/11091/chain-configs-should-move-to-toml-json-files
+
+	FallbackConfig = ChainSpecificConfig{
+		set:                              true,
+		EnableLegacyJobPipeline:          false,
+		EthBalanceMonitorBlockDelay:      1,
+		EthFinalityDepth:                 50,
+		EthGasBumpThreshold:              3,
+		EthGasBumpWei:                    *big.NewInt(5000000000), // 5 Gwei
```

Comment:
Are these ever meant to be nil ? I'd probably use non pointer members for structs unless we have a semantic purpose for nil.

Reply:
I agree